configuration/configuration.go

```diff
@@ -268,3 +264,174 @@ var camouflagedAccessToken = "751e16a8b39c0985066-AccessToken-4871777f2c13b32be8
+}
+
+// GetPostgresConnectionMaxRetries returns the number of times (as set via default, config file, or environment variable)
+// alm server will attempt to open a connection to the database before it gives up
+func (c *ConfigurationData) GetPostgresConnectionMaxRetries() int {
+	return c.postgresConnectionMaxRetries
+}
+
+// GetPostgresConnectionRetrySleep returns the number of seconds (as set via default, config file, or environment variable)
+// to wait before trying to connect again
+func (c *ConfigurationData) GetPostgresConnectionRetrySleep() time.Duration {
+	return c.postgresConnectionRetrySleep
+}
+
+// GetPopulateCommonTypes returns true if the (as set via default, config file, or environment variable)
+// the common work item types such as system.bug or system.feature shall be created.
+func (c *ConfigurationData) GetPopulateCommonTypes() bool {
+	return c.populateCommonTypes
+}
+
+// GetHTTPAddress returns the HTTP address (as set via default, config file, or environment variable)
+// that the alm server binds to (e.g. "0.0.0.0:8080")
+func (c *ConfigurationData) GetHTTPAddress() string {
+	return c.httpAddress
+}
+
+// IsPostgresDeveloperModeEnabled returns if development related features (as set via default, config file, or environment variable),
+// e.g. token generation endpoint are enabled
+func (c *ConfigurationData) IsPostgresDeveloperModeEnabled() bool {
+	return c.developerModeEnabled
+}
+
+// GetTokenPrivateKey returns the private key (as set via config file or environment variable)
+// that is used to sign the authentication token.
+func (c *ConfigurationData) GetTokenPrivateKey() []byte {
+	return c.tokenPrivateKey
+}
+
+// GetTokenPublicKey returns the public key (as set via config file or environment variable)
+// that is used to decrypt the authentication token.
+func (c *ConfigurationData) GetTokenPublicKey() []byte {
+	return c.tokenPublicKey
+}
+
+// GetGithubSecret returns the Github secret(as set via config file or environment variable)
+// that is used to make authorized Github API Calls.
+func (c *ConfigurationData) GetGithubSecret() string {
+	return c.githubSecret
+}
+
+// GetGithubClientID returns the Github Client ID(as set via config file or environment variable)
+// that is used to make authorized Github API Calls.
+func (c *ConfigurationData) GetGithubClientID() string {
+	return c.githubClientID
+}
+
+// GetGithubAuthToken returns the actual Github OAuth Access Token
+func (c *ConfigurationData) GetGithubAuthToken() string {
+	return c.githubAuthToken
+}
+
+// GetPostgresConfigString returns a ready to use string for usage in sql.Open()
+func (c *ConfigurationData) GetPostgresConfigString() string {
+	return fmt.Sprintf("host=%s port=%d user=%s password=%s DB.name=%s sslmode=%s",
+		getPostgresHost(),
+		getPostgresPort(),
+		getPostgresUser(),
+		getPostgresPassword(),
+		getPostgresDatabase(),
+		getPostgresSSLMode(),
+	)
+}
+
+type ConfigurationData struct {
+	// List of configuration items
```

Comment:
My approach would have been different:

```go
import "github.com/pkg/errors"

type ConfigurationData struct {
    v viper
}

func NewConfigurationData() (*ConfigurationData, error) {
    c := ConfigurationData{v: viper.New()}
    // Not sure if you want to reset your own viper.
    // c.v.Reset()
    c.v.SetConfigType("yaml")
    err := c.v.ReadInConfig() // Find and read the config file
    if err != nil {   // Handle errors reading the config file
        return nil, errors.Errorf("Fatal error config file: %s \n", err)
    }
    return &c, nil
}
```

Then you can have a function like this:

```go
func (c *ConfigurationData) GetGithubClientID() string {
 	return c.v.GetString(varGithubClientID)
 }
```

Then you can get rid of all the unexported functions and values are loaded on demand from a dedicated viper instance per `ConfigurationData` object.

See https://github.com/spf13/viper#viper-or-vipers for more information on how to have a dedicated viper instance.